### PR TITLE
add localized quotation marks on Notifications view #1422

### LINKED ISSUE #1422: [Localization] Localize the quotation marks for different languages
Presently Nos only supports quotation marks for English. This ticket is to enable the app to support localized quotation marks. 

AC
Localize quotation marks across the app specifically in:
- notes
- notifications 
- editor

### PR DESCRIPTION
**Pull request by bryanmontz**
## Issues covered
#1422 

## Description
This change localizes the quotation marks added around content in notes referenced on the Notifications view. It is implemented as a String extension that could be applied anywhere.

## How to test
1. Navigate to Notifications view.
2. Observe quotation marks around content.

## Screenshots
English:  
| Before | After |
| --- | --- |
|before|after|

French:  
french

Japanese:  
japanese

### CODE REVIEW

**Comment by joshuatbrown** [comment]
@bryanmontz Would you mind updating the PR description to remove "Closes #1422"? That wording breaks our process when the PR is merged since it closes the issue and moves it to Done, skipping our UAT and QA steps.

**Comment by bryanmontz** [comment]
> @bryanmontz Would you mind updating the PR description to remove "Closes #1422"? That wording breaks our process when the PR is merged since it closes the issue and moves it to Done, skipping our UAT and QA steps. If you just remove the word "Closes", everything in our process still works.

Done! Sorry about that.

**Review by joshuatbrown — CHANGES_REQUESTED** [review]
Code looks good. I did find an issue: when a link is at the beginning or end, the quote mark appears as part of the link. I'd expect the quote mark to always be white.

Screenshot 2024-08-26 at 9 55 21 AM

**Comment by bryanmontz** [comment]
Thanks for finding that! Fixed this issue. The only thing is that if `NoteParser` ever adds style attributes, we'll need them here as well. Right now it doesn't seem to.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Replaced hard-coded color values. [skip-release-notes]
 - Added a toggle for “Enable new media display” to Staging builds. [skip-release-notes]
 - Fixed typos in release notes. [skip-release-notes]
+- Localized the quotation marks on the Notifications view.
```
Comment (joshuatbrown) [inline CHANGELOG.md:20]: I think I'd like to have these sorted, so the ones that go in the release notes are above all the ones that say `[skip-release-notes]`. Would it make more sense to just have a section like `### Internal` under `## [Unreleased]`, and then it's implied that the Internal ones are skipped? Or two sections: `### External` and `### Internal`? Or maybe even better names than those -- I'm just not coming up with anything else right now.

This is just an idea; I don't think we need to do it right now unless we settle on a good approach, but I wanted to start talking about it.

**Review by joshuatbrown — APPROVED** [review]
Looks great, and works perfectly!

**Pull request merged**